build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/YamlRestCompatTestPlugin.java

```diff
@@ -55,6 +54,7 @@
     @Override
     public void apply(Project project) {
         final int compatibleVersion = Version.fromString(VersionProperties.getVersions().get("elasticsearch")).getMajor() - 1;
+        final String sourceSetName = "yamlRestTestV" + compatibleVersion + "Compat";
         final Path compatRestResourcesDir = Path.of("restResources").resolve("v" + compatibleVersion);
```

Comment:
If we are having version-specific source sets, do we still need to capture the version here in the file path as well. As it is now we have files like `src/yamlRestTestV7Compat/resources/rest-api-spec/test/v7compat/foo/bar.yml`. We probably don't need to track the version twice now. Also begs the question whether we need this change in the first place if we are _already_ managing per-version test resources. How will this affect backporting, etc?

Reply:
`src/yamlRestTestV7Compat/resources/rest-api-spec/test/v7compat/foo/bar.yml` uses a 4 part path to keept it from colliding with the actual v7 `test/foo/bar.yml` if it exists. (Both the real v7 tests and the custom test in this source set are on the same classpath, and there quite a bit of re-use of file names).  `v7compat` in the path is just a convention and included the v7 to help demark these are intended for compatibilty with v7. The fact this is just a convention and without any teeth (i.e. these would execute when master is v9) is a motivating factor for this PR. 

I can remove the `v7` from the `v7compat` convention in this PR since I agree it doesn't make sense now. I think I could also remove this extra directory entirely and ensure uniqueness of the files between the two sources of tests (with validation in Gradle). Thoughts ?

Since our REST compatilbity only supports 1 major version, these should never be backported and anything in this sourceset should be removed in the next major version.

Reply:
> I can remove the v7 from the v7compat convention in this PR since I agree it doesn't make sense now. I think I could also remove this extra directory entirely and ensure uniqueness of the files between the two sources of tests (with validation in Gradle). Thoughts ?

Yeah, I think that makes sense. It's just less stuff to try and keep in sync.